### PR TITLE
pkg/ipmi: remove buf.Bytes()[2] <= 0xFF check

### PR DESCRIPTION
**Pull request by binjip978**
buf.Bytes()[2] <= 0xFF are always true, wrap error and simplify error handling

### CODE REVIEW

File: pkg/ipmi/ipmi.go [hunk 1]
```diff
@@ -112,11 +113,7 @@ func (i *IPMI) ShutoffWatchdog() error {
 	data[5] = 0x0b // countdown msb - 5 mins
 
 	_, err := i.SendRecv(_IPMI_NETFN_APP, BMC_SET_WATCHDOG_TIMER, data[:6])
-	if err != nil {
```
Comment (rminnich) [inline pkg/ipmi/ipmi.go:115]: return i.SendRecv(...)?

File: pkg/ipmi/ipmi.go [hunk 2]
```diff
@@ -70,7 +71,7 @@ func (i *IPMI) RawSendRecv(msg Msg) ([]byte, error) {
 		case err == syscall.EINTR:
 			continue
 		case err != nil:
-			return nil, fmt.Errorf("ioctlSetReq failed with %v", err)
+			return nil, fmt.Errorf("ioctlSetReq failed with %w", err)
```
Comment (rminnich) [inline pkg/ipmi/ipmi.go:74]: "ioctlSetReq:%w may be the most common way to write this.

File: pkg/ipmi/ipmi.go [hunk 3]
```diff
@@ -140,8 +137,8 @@ func (e *Event) marshall() ([]byte, error) {
 		copy(data[3:16], buf.Bytes()[16:29])
 	}
 
-	// OEM non-timestamped
-	if buf.Bytes()[2] >= 0xE0 && buf.Bytes()[2] <= 0xFF {
```
Comment (rminnich) [inline pkg/ipmi/ipmi.go:144]: I wonder if that kind of test comes from C assumptions about promotion to int (the usual problem with C). Interesting. it's a trap we all fall into from time to time ...

**Review by rminnich — COMMENTED** [review]
for future people looking at commits:
pkg/ipmi, not pkg/impi :-)

thanks

File: pkg/ipmi/ipmi.go (hunk 1 above)
Comment (binjip978) [reply pkg/ipmi/ipmi.go:115]: Will not work i.SendRecv returns ([]byte, error) but LogSystemEvent return error.

File: pkg/ipmi/ipmi.go (hunk 2 above)
Comment (binjip978) [reply pkg/ipmi/ipmi.go]: updated

File: pkg/ipmi/ipmi.go (hunk 3 above)
Comment (binjip978) [reply pkg/ipmi/ipmi.go:144]: I was puzzled as well.

File: pkg/ipmi/ipmi.go (hunk 1 above)
Comment (rminnich) [reply pkg/ipmi/ipmi.go:115]: oops, what was I thinking.

**Review by rminnich — APPROVED** [review]

**Pull request merged**